code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1030,6 +1030,18 @@
 /mob/living/carbon/human/updatehealth()
 	. = ..()
 	dna?.species.spec_updatehealth(src)
+	if(HAS_TRAIT(src, TRAIT_IGNOREDAMAGESLOWDOWN))
+		remove_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN)
+		return
+	var/health_deficiency = max((maxHealth - health), staminaloss)
+	if(health_deficiency >= 40)
+		if(movement_type & FLYING)

```

Comment:
This seems really hacky. Shouldn't this be two different movespeed modifiers? One of them only applies to flying and the other doesn't apply to it.

Reply:
Something like
```
add_movespeed_modifier(id, override=TRUE, multiplicative_slowdown=(health_deficiency / 75, blacklisted_movetypes=FLOATING|FLYING)
add_movespeed_modifier(id, override=TRUE, multiplicative_slowdown=(health_deficiency / 25), movetypes=FLYING)
```
It should work